kratos/utilities/read_materials_utility.cpp

```diff
@@ -0,0 +1,265 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Marcelo Raschi
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/properties.h"
+#include "utilities/read_materials_utility.hpp"
+
+namespace Kratos
+{
+
+ReadMaterialsUtility::ReadMaterialsUtility(
+    Parameters rParameters,
+    Model& rModel
+    ) : mrModel(rModel)
+{
+    Parameters default_parameters(R"(
+    {
+        "Parameters" : {
+            "materials_filename" : "please specify the file to be opened"
+        }
+    }  )"
+    );
+
+    rParameters.RecursivelyValidateAndAssignDefaults(default_parameters);
+
+    // read json string in materials file, create Parameters
+    const std::string& materials_filename = rParameters["Parameters"]["materials_filename"].GetString();
+    std::ifstream infile(materials_filename);
```

Comment:
There is no error checking if the file exists, could be read ...

Reply:
True

Reply:
I added a check